test/integration/queries/pages/dehydrated-state.tsx

```diff
@@ -11,7 +11,7 @@ import {Suspense} from "react"
 
 export const getServerSideProps: GetServerSideProps = async (ctx) => {
   const queryClient = new QueryClient()
-  const queryKey = getQueryKey(getDate, null)
+  const queryKey = getQueryKey(getDate, undefined)
```

Comment:
Interesting suggestion, I like it. It took me a while to figure it out and there was a small piece in the react-query docs that made me realize there was a mismatch.

Anyway we can make it easier for users to form the right code I am in support of :)

Reply:
Yeah I knew from the start we needed to make this better.

How about this?

```ts
import {prefetchQuery} from 'blitz'

export const getServerSideProps = async ctx => {

  await prefetchQuery(getData, undefined)

  return { props: {} }
}
```

And then we automatically add `new QueryClient()`, `dehydrate` etc via a babel plugin.

Reply:
A small suggestion would be to require that `dehydratedState` be passed into `props: {}` to avoid the edge case of a user declaring a prop `dehydratedState` and having overridden without them knowing. Perhaps we could rename the prop to `prefetchedQuery` for clarity and then transform in `blitz-root`?

```
export const getServerSideProps = async ctx => {
  const dehydratedState = await prefetchQuery(getData, undefined)
  return { props: {
    dehydratedState
  } }
}

Reply:
The babel plugin can throw a helpful error or something in that case, right? I think having `dehydratedState` exposed is exposing unnecessary implementation details.